api/processors/actions/group/finish.py

```diff
@@ -55,8 +57,22 @@ def _set_teachers_status(self) -> None:
         )
 
     def _set_students_status(self) -> None:
-        self.group.students.update(
+        annotated_students = Student.objects.annotate(groups_count=Count("groups")).filter(
+            groups=self.group
+        )
+
+        # TODO: We should check group status or something
```

Comment:
@lemontree210 I need your help here :)

Reply:
When a group gets finished or aborted, all students are moved to its  `.students_former`, and this group will be in the student's `.groups_former`. This should mean that `.groups` only shows groups that are either working or awaiting start, so I don' think any additional check of group status is needed.

Reply:
Another thing is that it is strange from the point of readability when we set status `AWAITING_DECISION` for students that have `groups_count=1`. I know that this is because this method is called before the students are moved from `.students` to `.students_former`. But I like the way `._set_teachers_status()` is written in that it is clear that if a teacher has no other groups (not "if teacher has 1 group"), then a certain status should be set.

Of course, you can simply add a comment saying that 
```
# after moving the student, this 1 will become 0
```

Reply:
Let's just add a comment to be consistent with other solutions :)